### PR TITLE
fix(test): Fix test assertions for Green Line suspension data

### PR DESCRIPTION
**Pull request by arkadyan**
#### Summary of changes
No ticket, fixes currently failing tests

- fix: Properly handle different valid input types
- fix(test): Improve tests to match descriptions
- fix(test): Fix test assertions for Green Line suspension data
- fix(test): Fix more test data

~~Note: I left that last commit separate because those failures seem more suspicious to me. I'm going to keep looking into them as I work through the GL diagram issues.~~
Given that the line diagram is working better with the latest GTFS data, I'm actually probably less worried about those failures than I was at first glance.

---

Before getting review, please check the following:

* [x] Does frontend functionality render and work correctly in IE?
* [x] Have we load-tested any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../apps/site/load_tests/README.md)
* [x] Are interactive elements accessible to screen readers?
* [x] Have you checked for tech debt you can address in the area you're working in?
* [x] If this change involves routes, does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?
* [x] Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?

### CODE REVIEW

File: apps/site/test/site_web/controllers/schedule/finder_api_test.exs [hunk 1]
```diff
@@ -295,12 +295,14 @@ defmodule SiteWeb.ScheduleController.FinderApiTest do
         |> json_response(200)
 
       assert %{"times" => times} = trip
-      assert length(times) == 8
+      destination_stop_ids = Enum.map(times, &get_in(&1, ["schedule", "stop", "id"]))
```
Comment (thecristen) [inline apps/site/test/site_web/controllers/schedule/finder_api_test.exs:298]: I always forget about `get_in`!

File: apps/site/lib/site_web/views/schedule_view.ex [hunk 2]
```diff
@@ -362,22 +362,24 @@ defmodule SiteWeb.ScheduleView do
     "/fares/" <> route_type <> "-fares"
   end
 
-  @spec single_trip_fares(Route.t()) :: [{String.t(), String.t() | iolist}]
+  @spec single_trip_fares(Route.t() | Route.type_int() | String.t()) :: [
```
Comment (thecristen) [inline apps/site/lib/site_web/views/schedule_view.ex:365]: Everywhere this is used seems to be passing in a `%Route{}`, I'm curious why change this?

**Review by thecristen — APPROVED** [review]
LGTM with one question.

File: apps/site/lib/site_web/views/schedule_view.ex (hunk 2 above)
Comment (arkadyan) [reply apps/site/lib/site_web/views/schedule_view.ex:365]: The immediate reason was the [test case](https://github.com/mbta/dotcom/blob/master/apps/site/test/site_web/views/schedule_view_test.exs#L449) for this function calls it using a string and was breaking due to a recent change. But it does seem like this function is only called from a single place, so I'm ok with changing the test instead and leaving the more restrictive spec.

Comment (arkadyan) [reply apps/site/lib/site_web/views/schedule_view.ex]: [Fixed](https://github.com/mbta/dotcom/pull/1396/commits/0d41fa2ef55aed55e4f63c68d1ea37a51f3c2092)

**Pull request merged**